platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/delegates/TaskContainerDelegate.kt

```diff
@@ -76,27 +76,43 @@ abstract class TaskContainerDelegate : TaskContainer {
     override fun addAllLater(provider: Provider<out Iterable<Task>>) =
         delegate.addAllLater(provider)
 
+    @Deprecated("Use the right `register` variation instead.", ReplaceWith("register(name)"))
```

Comment:
Looking for feedback on how best to do those annotations for Kotlin

Reply:
I think the convention is to use `this`. Not sure if it makes semantic difference, but that's how they do it in the Kotlin  stdlib.

```suggestion
    @Deprecated("Use the right `register` variation instead.", ReplaceWith("this.register(name)"))
```

However, I'd have a separate consideration of whether providing a quick fix is the right idea. It will change semantics from eager to lazy, and in some cases like this  `create(optionsMap)` the quick fix would actually lose information.

I'd say if there is one-to-one method mapping from a `create` overload to the `register` overload it's okay to provide the quick fix, but for other cases we should explain what needs to be done in the javadoc

Reply:
Interesting, we are not using `this` in other places where we have such an annotation. And I wonder what it means from a delegate.

I will not change this for now.

Regarding the validity of the `ReplaceWith` for the map notation, I agree. Removing them.

Reply:
After some testing, `this` was mandatory in some places, it was added there. Where it was not, I left it out.

Reply:
What makes `this` mandatory? Is there an official documentation about the semantics of `ReplaceWith` with respect to `this` being used or not?

Reply:
`tasks.create(...)` and `ReplaceWith` without `this` would cause the replacement to become `register(...)` (Note the `tasks` disappeared.

That made `this` mandatory in those cases.